src/Raven.Server/Utils/ThreadIoStatsReader.cs

```diff
@@ -0,0 +1,214 @@
+            public double MbTotal;
+
+            // Split metrics when available
+            public double? ReadOpsPerSecLast;
+            public double? WriteOpsPerSecLast;
+            public double? ReadMbPerSecLast;
+            public double? WriteMbPerSecLast;
+            public long? ReadOpsTotal;
+            public long? WriteOpsTotal;
+            public double? ReadMbTotal;
+            public double? WriteMbTotal;
+        }
+
+        private ThreadIoStatsReader()
+        {
+            if (PlatformDetails.RunningOnLinux == false)
+                return;
+
+            try
+            {
+                var ravenDebugExec = Path.Combine(AppContext.BaseDirectory,
+                    PlatformDetails.RunningOnPosix ? "Raven.Debug" : "Raven.Debug.exe");
+
+                if (File.Exists(ravenDebugExec) == false)
+                {
+                    if (Logger.IsInfoEnabled)
+                        Logger.Info($"Raven.Debug not found at '{ravenDebugExec}', thread IO stats will be disabled.");
+                    return;
+                }
+
+                using (var currentProcess = Process.GetCurrentProcess())
+                {
+                    var psi = new ProcessStartInfo
+                    {
+                        FileName = ravenDebugExec,
+                        Arguments = $"threads-io --pid {currentProcess.Id} --interval 1000",
+                        RedirectStandardOutput = true,
+                        RedirectStandardError = true,
+                        UseShellExecute = false,
+                        CreateNoWindow = true
+                    };
+
+                    _process = new Process { StartInfo = psi, EnableRaisingEvents = true };
+                    _process.OutputDataReceived += (s, e) =>
+                    {
+                        if (e.Data == null)
+                            return;
+                        try
+                        {
+                            using var doc = JsonDocument.Parse(e.Data);
+                            var root = doc.RootElement;
+                            if (root.TryGetProperty("tid", out var tidEl) == false)
+                                return;
+                            var tid = tidEl.GetInt32();
+                            var opsPerSec = root.GetProperty("opsPerSec").GetDouble();
+                            var mbPerSec = root.GetProperty("mbPerSec").GetDouble();
+                            var opsTotal = root.GetProperty("opsTotal").GetInt64();
+                            var mbTotal = root.GetProperty("mbTotal").GetDouble();
+
+                            double? readOpsPerSec = null, writeOpsPerSec = null, readMbPerSec = null, writeMbPerSec = null;
+                            long? readOpsTotal = null, writeOpsTotal = null;
+                            double? readMbTotal = null, writeMbTotal = null;
+
+                            if (root.TryGetProperty("opsReadPerSec", out var opsReadPerSecEl))
```

Comment:
"opsReadPerSec" and others - those are generated by Raven.Debug, right? Can we use `nameof` or deserialize entire object? It's well defined (by us) JSON structure

Reply:
Would be _much_ better to just use our own deserialization, rather than System.Text.Json